packages/util/src/obj.ts

```diff
@@ -15,135 +15,133 @@
  * limitations under the License.
  */
 
-// See http://www.devthought.com/2012/01/18/an-object-is-not-a-hash/
+type Keys<T> = Extract<keyof T, string>;
+type Values<T> = T[keyof T];
 
-export const contains = function(obj, key) {
+export function contains<T extends object>(obj: T, key: keyof T): boolean {
   return Object.prototype.hasOwnProperty.call(obj, key);
-};
+}
 
-export const safeGet = function(obj, key) {
-  if (Object.prototype.hasOwnProperty.call(obj, key)) return obj[key];
-  // else return undefined.
-};
+export function safeGet<T extends object, K extends keyof T>(
+  obj: T,
+  key: K
+): T[K] | undefined {
+  if (Object.prototype.hasOwnProperty.call(obj, key)) {
+    return obj[key];
+  } else {
+    return undefined;
+  }
+}
 
 /**
  * Enumerates the keys/values in an object, excluding keys defined on the prototype.
  *
- * @param {?Object.<K,V>} obj Object to enumerate.
- * @param {!function(K, V)} fn Function to call for each key and value.
- * @template K,V
+ * @param obj Object to enumerate.
+ * @param fn Function to call for each key and value.
  */
-export const forEach = function(obj, fn) {
-  for (var key in obj) {
-    if (Object.prototype.hasOwnProperty.call(obj, key)) {
-      fn(key, obj[key]);
-    }
+export function forEach<T extends object>(
+  obj: T,
+  fn: (key: Keys<T>, value: Values<T>) => void
+) {
+  for (const key of Object.keys(obj)) {
+    // Object.keys() doesn't return an Array<keyof T>
+    // https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208
+    fn(key as Keys<T>, obj[key]);
   }
-};
+}
 
 /**
  * Copies all the (own) properties from one object to another.
- * @param {!Object} objTo
- * @param {!Object} objFrom
- * @return {!Object} objTo
  */
-export const extend = function(objTo, objFrom) {
-  forEach(objFrom, function(key, value) {
-    objTo[key] = value;
-  });
-  return objTo;
-};
+export function extend<T extends object, U extends object>(
+  objTo: T,
+  objFrom: U
+): T & U {
+  return Object.assign(objTo, objFrom);
+}
 
 /**
  * Returns a clone of the specified object.
- * @param {!Object} obj
- * @return {!Object} cloned obj.
+ * @return cloned obj.
  */
-export const clone = function(obj) {
-  return extend({}, obj);
-};
+export function clone<T extends object>(obj: T): T {
```

Comment:
If this is not used often, this is a good candidate for removal.

Reply:
Done in next PR.